### DISCUSSION TITLE
Using External Api to Update WooCommerce product stock

### DISCUSSION THREAD

**Discussion by darkslayert** [woocommerce/woocommerce#47833] [General]
Hi,
I have an accounting software, which has a rest API available. I want the information in this database to update my WooCommerce store (at least twice a day) – specifically to keep things like product quantity up-to-date and synced.
Everything I’ve read so far is about WooCommerce sending info over to external databases – not the other way around.
In other words, I want product inventory to be controlled by an external database that then pushes/updates WooCommerce a couple of times a day.
Can someone help me understand the steps needed to set this up? I know I have rest API’s on both sides (with the external database AND with woocommerce), but I’m not sure what to do with them.
Thank you,

**Comment by Dexter-0909** [comment]
This is the same question I was going to ask, is it solved yet?

**Comment by barryhughes** [comment]
So, *probably,* you will need to create a custom application of some kind to act as a sort of glue layer, and to hold whatever logic you might need to reconcile both systems. That might be a standalone app or script, or it could be implemented as a WordPress plugin, or something else.

> Everything I’ve read so far is about WooCommerce sending info over to external databases

I'm not quite sure what you were looking at ([webhooks](https://woocommerce.com/document/webhooks/), possibly?), but I'd refer you here to our REST API documentation:

https://woocommerce.github.io/woocommerce-rest-api-docs/#introduction

This allows you to read data (such as inventory levels) and also to write data (you can change inventory levels). Let me know if that helps, or if you have some more specific questions, and if we can steer you in the right direction we will 🙂 ... remember too that it is possible to [hire experienced developers](https://woocommerce.com/vendor/codeable/) who can help you figure out what's needed in your specific case.
